esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,381 @@
+        filled in for both the vm and privileges tables.
+        """
+
+        # Create the entry in the tenants table
+        tenant = DockerVolumeTenant(name, description, default_datastore,
+                                    default_privileges, vms, privileges)
+        id = tenant.id
+        self.conn.execute(
+            "insert into tenants(id, name, description, default_datastore) values (?, ?, ?, ?)",
+            (id, name, description, default_datastore)
+        )
+
+        # Create the entries in the vms table
+        vms = [(vm_id, vm_name, id) for (vm_id, vm_name) in vms]
+        #print vms
+        if vms:
+            self.conn.executemany(
+              "insert into vms(vm_id, vm_name, tenant_id) values (?, ?, ?)",
+              vms
+            )
+
+        # Create the entries in the privileges table
+        # TODO: data scrubbing, ensure all keys exist
+        for p in default_privileges:
+            p['tenant_id'] = id
+        self.conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            default_privileges
+        )
+
+        for p in privileges:
+            p['tenant_id'] = id
+        self.conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+        self.conn.commit()
+        return tenant
+    
+    def list_tenants(self):
+        """
+            Return a list of DockerVolumeTenants objects.
+        """
+        tenant_list = []
+        cur = self.conn.execute(
+           "select * from tenants"
+        )
+        result = cur.fetchall()
+       
+        for r in result:
+            # loop through each tenant
+            id = r['id']
+            name = r['name']
+            description = r['description']
+            default_datastore = r['default_datastore']
+            
+            # search vms for this tenant
+            vms = []
+            cur = self.conn.execute(
+                "select * from vms where tenant_id=?",
+                (id,)
+            )
+            vms = cur.fetchall()
+            
+            # search privileges and default_privileges for this tenant
+            privileges = []
+            cur = self.conn.execute(
+                "select * from privileges where tenant_id=? and datastore!=?",
+                (id,default_datastore)    
+            )
+            privileges = cur.fetchall()
+
+            default_privileges = []
+            cur = self.conn.execute(
+                "select * from privileges where tenant_id=? and datastore=?",
+                (id,default_datastore)    
```

Comment:
please, do add space after comma - here and in other places. See  https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements for more details.
Same in SQL expresisons (spaces around = and !=). Also it's better to use `==` not `=` (same result but reads easier)